OSSM-6063 [DOC] OpenShift Service Mesh Console (OSSMC) plugin Integration (#88446)
openshift/openshift-docs: observability/kiali/ossm-console-plugin.adoc

:_mod-docs-content-type: ASSEMBLY
[id="ossm-console-plugin"]
= Using OpenShift Service Mesh Console plugin
include::_attributes/common-attributes.adoc[]
:context: ossm-console-plugin

toc::[]

The {SMPlugin} extends the {ocp-short-name} web console with a {SMProductShortName} menu and enhanced tabs for workloads and services.

[IMPORTANT]
====
If your browser does not trust the certificate, you can manually accept it in the Kiali standalone UI before accessing the {SMPluginShort}.
====

include::modules/ossm-about-console-plugin.adoc[leveloffset=+1]

[id="ossm-install-console-plugin_{context}"]
== Installing {sm-plugin-full}

> You can install the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}. In most cases, you must install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.

Reviewer: I think the installation of  the latest version of the Kiali Operator is more a recommendation than a must. There is no problem if the customer installs a previous version of the Operator to install a previous version of OSSMC, only that maybe it does not contain the latest z-stream version of OSSMC.

Reviewer 2: Right... `"must"` makes it seem like the customer has no choice. But that technically isn't accurate. I would say `"should"`

```suggestion
You can install the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}. In most cases, you should install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.
```

You can still install a previous version of OSSMC using a previous version of Kiali Operator - you just won't get the latest z-stream release (which is mentioned here in these docs already). Why would someone want to do this? Perhaps there is a bug in the latest Kiali Operator that the customer needs to temporarily workaround in order to install anything.

Author: I am using 'can' instead of 'should' to follow doc guidelines.

:_mod-docs-content-type: ASSEMBLY
[id="ossm-console-plugin"]
= Using OpenShift Service Mesh Console plugin
include::_attributes/common-attributes.adoc[]
:context: ossm-console-plugin

toc::[]

> The {SMPlugin} extends the {ocp-short-name} web console with a {SMProductShortName} menu and enhanced tabs for workloads and services.

Reviewer 3: ```suggestion
The {SMPlugin} extends the {ocp-product-title} web console with a {SMProductShortName} menu and enhanced tabs for workloads and services.
```

include::modules/ossm-about-console-plugin.adoc[leveloffset=+1]

[id="ossm-install-console-plugin_{context}"]
== Installing {sm-plugin-full}

You can install the {SMPluginShort} with the Kiali Operator by creating a `OSSMConsole` resource with the corresponding plugin settings. It is recommended to install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.

.OSSM version compatibility
[options="header"]
|===
|OSSM version |Kiali Server version |OSSMC plugin version |OCP version

|3.0 |v2.4 |v2.4 |4.15+

|2.6 |v1.73 |v1.73 |4.15-4.18

|2.5 |v1.73 |v1.73 |4.14-4.18
|===

[NOTE]
====

> {SMPluginShort} is only supported on {ocp-short-name} 4.15 and above. For {ocp-short-name} 4.14 users, only the standalone Kiali console is accessible.

Reviewer 3: ```suggestion
{SMPluginShort} is only supported on {ocp-product-title} 4.15 and above. For {ocp-product-title} 4.14 users, only the standalone Kiali console is accessible.
```

> You can install the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}. In most cases, you can install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.
> 
> You can install the {SMPluginShort} with Kiali Operator by craeting a `OSSMConsole` resource with the corresponding plugin settings.
> 
> .OSSM version compatibility
> [options="header"]
> |===
> |OSSM version |Kiali Server version |OSSMC plugin version |OCP version
> 
> |3.0 |v2.4 |v2.4 |4.15+
> 
> |2.6 |v1.73 |v1.73 |4.15-4.18
> 
> |2.5 |v1.73 |v1.73 |4.14-4.18
> |===
> 
> [NOTE]
> ====
> {SMPluginShort} is only supported on {ocp-short-name} 4.15 and above. For {ocp-short-name} 4.14 users, only the standalone Kiali console is accessible.
> ====

Reviewer: I think therea are too many statements that start with `you can install`. I have reordered and rephrased them a bit. 
```suggestion
You can install the {SMPluginShort} with the Kiali Operator by creating a `OSSMConsole` resource with the corresponding plugin settings. It is preferred to install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.

.OSSM version compatibility
[options="header"]
|===
|OSSM version |Kiali Server version |OSSMC plugin version |OCP version

|3.0 |v2.4 |v2.4 |4.15+

|2.6 |v1.73 |v1.73 |4.15-4.18

|2.5 |v1.73 |v1.73 |4.14-4.18
|===

[NOTE]
====
{SMPluginShort} is only supported on {ocp-short-name} 4.15 and above. For {ocp-short-name} 4.14 users, only the standalone Kiali console is accessible.
====

You can install the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}.
```

> You can install the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}.

Reviewer 3: Ah - apparently the `oc-first` attribute isn't in the service mesh attributes file. If you want to add it to your file for use, the format is a little weird:

```
:oc-first: pass:quotes[OpenShift CLI (`oc`)]
```

Otherwise, you can just type out the thing when you need it (I find that annoying to type, hence created the attribute :wink:)
```suggestion
You can install the {SMPluginShort} by using the {ocp-product-title} web console or the OpenShift CLI (`oc`).
```

Author: Thanks @jeana-redhat! I removed the attribute because it wasn't displaying correctly. I will try adding the attribute and see what happens again. 😄

Reviewer 3: it's working, yay :tada:

include::modules/ossm-install-console-plugin-ocp-web-console.adoc[leveloffset=+2]

include::modules/ossm-install-console-plugin-ocp-cli.adoc[leveloffset=+2]

:_mod-docs-content-type: PROCEDURE
[id="ossm-uninstall-console-plugin_{context}"]
== Uninstalling {sm-plugin-full}

> You can uninstall the {SMPluginShort} by using the {ocp-short-name} web console or the {ocp-short-name} CLI.

Reviewer 3: ```suggestion
You can uninstall the {SMPluginShort} by using the {ocp-product-title} web console or the {oc-first}.
```

[id="ossm-install-console-plugin_{context}"]
== Installing {sm-plugin-full}

> You can install the {SMPluginShort} with the Kiali Operator by creating a `OSSMConsole` resource with the corresponding plugin settings. It is preferred to install the latest version of the Kiali Operator, even while installing a previous {SMPluginShort} version, as it includes the latest z-stream release.

Reviewer: @shreyasiddhartha @mazz is the word `preferred` okay here, or is `recommended` better? In Spanish, both words are mostly interchangeable, but maybe there is a different context in English.

Author: IMO, `recommended` suits better. So, i am changing it.
